tests/Unit/Helpers/DataStructures/VectorImplTestHelper.hpp

```diff
@@ -44,15 +44,20 @@ void vector_test_construct_and_assign(
   MAKE_GENERATOR(gen);
   UniformCustomDistribution<tt::get_fundamental_type_t<ValueType>> dist{low,
                                                                         high};
-  static_assert(default_vector_impl_static_size >= 2 and
-                default_vector_impl_static_size <= 20);
-  UniformCustomDistribution<size_t> static_sdist{
-      2, default_vector_impl_static_size};
-  UniformCustomDistribution<size_t> sdist{default_vector_impl_static_size + 1,
-                                          20};
-
-  std::vector<size_t> sizes{static_sdist(gen), default_vector_impl_static_size,
-                            sdist(gen)};
+  std::vector<size_t> sizes;
+  if constexpr (default_vector_impl_static_size >= 2) {
```

Comment:
Instead of checking the default size, check the size for `VectorType` and call the test functions somewhere with a vector with non-zero static size.